pkg/node/address.go

```diff
@@ -199,12 +199,16 @@ func InitBPFMasqueradeAddrs(devices []string) error {
 	if option.Config.EnableIPv4 {
 		ipv4MasqAddrs = make(map[string]net.IP, len(devices))
 		for _, device := range devices {
-			ip, err := firstGlobalV4Addr(device, nil, preferPublicIP)
+			ip, err := firstGlobalV4Addr(device, GetK8sNodeIP(), preferPublicIP)
 			if err != nil {
 				return fmt.Errorf("Failed to determine IPv4 of %s for BPF masq", device)
 			}
 
 			ipv4MasqAddrs[device] = ip
+			log.WithFields(logrus.Fields{
+				"ip":     ip,
+				"device": device,
```

Comment:
Consider using / extending the `logfields` package for these fields.

Reply:
done